llvm/lib/Target/AIE/AIEMachineScheduler.cpp

```diff
@@ -426,6 +429,32 @@ static unsigned getMinSchedulableCycle(SchedBoundary &Zone) {
   return MinSchedulableCycle;
 }
 
+bool AIEPostRASchedStrategy::doesNotProgressInZone(const SchedBoundary &Zone,
+                                                   const SUnit &SU) const {
+  // If SU is a fixed instruction in the other zone, it isn't available.
+  if (isFixedSU(SU, /*IsTop*/ !Zone.isTop()))
```

Comment:
Nit: I find `if (isFixedSU(SU, !Zone.isTop()))` just as clear, especially with the comment above.